src/storage/storages.rs

```diff
@@ -83,12 +96,40 @@ unsafe impl<T> DistinctStorage for HashMapStorage<T> {}
 ///
 /// Note that this only stores the data (`T`) densely; indices
 /// to the data are stored in a sparse `Vec`.
+///
+/// `as_slice()` and `as_mut_slice()` indices are local to this
+/// `DenseVecStorage` at this particular moment. These indices
+/// cannot be compared with indices from any other storage, and
+/// a particular entity's position within this slice may change
+/// over time.
 #[derive(Derivative)]
 #[derivative(Default(bound = ""))]
 pub struct DenseVecStorage<T> {
     data: Vec<T>,
-    entity_id: Vec<Index>,
-    data_id: Vec<Index>,
+    entity_id: Vec<MaybeUninit<Index>>,
```

Comment:
Does `entity_id` need to contain `MaybeUninit`? We don't `set_len` on it, right?

Reply:
Oops, got over-zealous.